openedx/xapi-db-load
# Bundles xapi-db-load into a command-line tool with setuptools

...and handles a [breaking change with query settings from `clickhouse-connect==0.5.0`](https://github.com/ClickHouse/clickhouse-connect/blob/main/CHANGELOG.md#warning----breaking-change----removing-get_client-arbitrary-keyword-arguments).

**Testing instructions**

E.g inside a docker container with access to `clickhouse_service` from https://github.com/bmtcril/tutor-contrib-clickhouse :

1. Checkout this branch, and change into the repo directory.
1. In a new python virtualenv, run `pip install .` to install it and the latest dependencies.
1. Run `xapi-db-load --backend clickhouse...` to ensure that the script still runs as expected.

**Author Notes & Concerns**

1. @bmtcril Can you tag this as `0.1` after merging? Then I can reference it in my tutor init scripts, and we don't have to worry about api changes on the script.
